src/main/java/mate.academy/lib/Injector.java

```diff
@@ -1,13 +1,82 @@
 package mate.academy.lib;
 
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.util.HashMap;
+import java.util.Map;
+import mate.academy.service.FileReaderService;
+import mate.academy.service.ProductParser;
+import mate.academy.service.ProductService;
+import mate.academy.service.impl.FileReaderServiceImpl;
+import mate.academy.service.impl.ProductParserImpl;
+import mate.academy.service.impl.ProductServiceImpl;
+
 public class Injector {
     private static final Injector injector = new Injector();
+    private Map<Class<?>, Object> instances;
+    private Map<Class<?>, Class<?>> interfaceImpl;
+
+    public Injector() {
+        instances = new HashMap<>();
+        interfaceImpl = Map.of(
+                FileReaderService.class, FileReaderServiceImpl.class,
+                ProductParser.class, ProductParserImpl.class,
+                ProductService.class, ProductServiceImpl.class
+        );
+    }
```

Comment:
```suggestion
    private Map<Class<?>, Object> instances = new HashMap<>();
    private Map<Class<?>, Class<?>> implementaionMap = Map.of(
                FileReaderService.class, FileReaderServiceImpl.class,
                ProductParser.class, ProductParserImpl.class,
                ProductService.class, ProductServiceImpl.class
        );
```

It will make code more readable and iff you plan to add more interfaces and their implementations, then the constructor with Map.of() might become unwieldy and hard to manage.

Reply:
`implementaionMap` - will be more understandable